src/common/ceph_argparse.cc

```diff
@@ -123,19 +110,21 @@ void env_to_vec(std::vector<const char*>& args, const char *name)
   }
   g_str_vec_lock.unlock();
 
-  vector<string>::iterator i;
-  for (i = g_str_vec.begin(); i != g_str_vec.end(); ++i)
-    env.push_back(i->c_str());
-  if (split_dashdash(env, env_options, env_arguments))
-    dashdash = true;
+  std::vector<const char*> env;
+  for (const auto& s : g_str_vec) {
+    env.push_back(s.c_str());
+  }
+  auto [env_options, env_arguments] = split_dashdash(env);
```

Comment:
As commented in the other PR, to me it doesn't really make sense to support `--` in `CEPH_ARGS`. There is no Ceph "global" positional argument that would apply to all CLI tools.

Reply:
i think we need to avoid introducing a breaking change in a fix which will be backported. we can always do this in master.